mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
+  Product.findById(id) 
+    .populate('products.product', 'name price')
+    .exec((err, product) => { 
+      if (err || !product) {
```

Comment:
The condition `if (err || !product)` is not handling the error properly. It should be `if (err) { ... } else if (!product) { ... }` to provide a more specific error message for each scenario.